### ISSUE TITLE
API: Redirect after OAuth login when auth.anonymous is enabled

### ISSUE THREAD

**Issue by papagian** [grafana/grafana#22709]
**What happened**:
WIth `auth.anonymous` enabled (and after having logged out):

- Browse dashboard X
- Click on login in sidemenu and redirect to login page
- Login using OAUth
The user is not redirected to dashboard X

**What you expected to happen**:
The user should be redirected to dashboard X

**How to reproduce it (as minimally and precisely as possible)**:
```
docker run \
-e GF_AUTH_ANONYMOUS_ENABLED=true \
-e GF_AUTH_GENERIC_OAUTH_ENABLED=true \
-e GF_AUTH_GENERIC_OAUTH_NAME=okta \
-e GF_AUTH_GENERIC_OAUTH_CLIENT_ID=<client_id> \
-e GF_AUTH_GENERIC_OAUTH_CLIENT_SECRET=<client_secreat> \
-e GF_AUTH_GENERIC_OAUTH_SCOPES="openid profile email" \
-e GF_AUTH_GENERIC_OAUTH_AUTH_URL=<auth_url> \
-e GF_AUTH_GENERIC_OAUTH_TOKEN_URL=<token_url> \
-e GF_AUTH_GENERIC_OAUTH_API_URL=<api_url>  \
--rm -p 3000:3000 grafana/grafana:<grafana_version>
```

**Anything else we need to know?**:

**Environment**:
- Grafana version:
6.5.2 and 6.6.2

- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by sparse91** [comment]
This problem still exists in 7.4.

**Comment by marefr** [comment]
@papagian do you remember if  #22285 solved this?

**Comment by papagian** [comment]
> @papagian do you remember if #22285 solved this?

No, I believe it is solved by #22709
I have tested it and the issue exists in 6.7.0-beta1 but it's resolved in 7.1.0-beta1.
